### DISCUSSION TITLE
Suggestion: Changes to system-icons.css (solving known color bug)

### DISCUSSION THREAD

**Discussion by ochi12** [rafaelmardojai/firefox-gnome-theme#857] [General]
machine: Ubuntu
I used an online calculator for svg filter #808080 as the base color, #808080 (note this is from ubuntu's Yaru icon)

`I'm not sure if other icon themes uses #808080 as base color for their symbolic icons. `

![image](https://github.com/user-attachments/assets/58485086-25e7-482b-bdc0-7258aad1c79b)
![image](https://github.com/user-attachments/assets/b1f1a7d6-2354-4faf-a0c2-34aa7547dc8a)

The suspect in this snippet is the extra 'filter:' word
`--gnome-icons-hack-filter: filter: ` is supposed to be `--gnome-icons-hack-filter:  `
```
/* Icons light/dark fix coloring hack filters */
:root {
	--gnome-icons-hack-filter: filter: brightness(0) saturate(100%) invert(37%) sepia(8%) saturate(683%) hue-rotate(183deg) brightness(95%) contrast(84%);
	--gnome-window-icons-hack-filter: filter: brightness(0) saturate(100%) invert(37%) sepia(8%) saturate(683%) hue-rotate(183deg) brightness(95%) contrast(84%);
}
@media (prefers-color-scheme: dark) {
	:root {
		--gnome-icons-hack-filter: brightness(0) saturate(100%) invert(94%) sepia(11%) saturate(362%) hue-rotate(184deg) brightness(95%) contrast(87%);
		--gnome-window-icons-hack-filter: brightness(0) saturate(100%) invert(94%) sepia(11%) saturate(362%) hue-rotate(184deg) brightness(95%) contrast(87%);
	}
}
```

Updated Snipper
```
/* Icons light/dark fix coloring hack filters */
:root {
	--gnome-icons-hack-filter: brightness(0) saturate(100%) invert(0%) sepia(5%) saturate(7500%) hue-rotate(295deg) brightness(104%) contrast(104%);
	--gnome-window-icons-hack-filter: filter: brightness(0) saturate(100%) invert(37%) sepia(8%) saturate(683%) hue-rotate(183deg) brightness(95%) contrast(84%);
}
@media (prefers-color-scheme: dark) {
	:root {
		--gnome-icons-hack-filter:  brightness(0) saturate(100%) invert(100%) sepia(100%) saturate(0%) hue-rotate(164deg) brightness(101%) contrast(101%);
		--gnome-window-icons-hack-filter: brightness(0) saturate(100%) invert(94%) sepia(11%) saturate(362%) hue-rotate(184deg) brightness(95%) contrast(87%);
	}
}
```

checking in light theme (ICONS are now black) instead of gray
![image](https://github.com/user-attachments/assets/873a80e9-9a95-4181-a8bc-f9a288c36596)

**Comment by rafaelmardojai** [comment]
Thanks, I have corrected the typo in the code. But sadly that doesn't have anything to do with the known color bug. The problem is that a different filter might be needed depending on the icon theme, so this will only work for Yaru.

**Reply by ochi12** [reply]
> so this will only work for Yaru.


Well then additional options in the about:config with help, like choosing Icon theme. Specifying the icon theme  with off course string value instead of boolean or will also set the filter somewhere.
